packages/flutter_tools/lib/src/base/logger.dart

```diff
@@ -221,7 +221,11 @@ class AnsiTerminal {
   String clearScreen() => supportsColor ? _clear : '\n\n';
 
   set singleCharMode(bool value) {
-    stdin.lineMode = !value;
+    try {
+      stdin.lineMode = !value;
+    } on IOException {
+      // This can throw for some terminals; we ignore the error.
```

Comment:
As I mentioned over email, this is not a regression, but a bug-fix. Previously we were swallowing the errors from tc{get,set}attr().